pyccel/stdlib/ndarrays/ndarrays.c

```diff
@@ -42,12 +42,14 @@ t_ndarray   array_create(int32_t nd, int64_t *shape,
         case nd_bool:
             arr.type_size = sizeof(bool);
             break;
+        #ifndef __NVCC__
         case nd_cfloat:
             arr.type_size = sizeof(float complex);
             break;
         case nd_cdouble:
             arr.type_size = sizeof(double complex);
             break;
+        #endif
```

Comment:
@EmilyBourne Complex are not supported for now as we should handle them in a special way

Reply:
the macro is set to true when compiling with the nvidia cuda compiler `nvcc` where we do not have the support of `float complex` and `double complex` and we should use `#include <cuComplex.h>` instead when compiling with nvcc.

Reply:
`nvcc` is focused on CUDA programming and is compatible with all NVIDIA GPU architectures.
`nvc` is designed for HPC applications and is optimized for the latest NVIDIA GPU architectures.

Reply:
According to [Nvidia's docs](https://docs.nvidia.com/hpc-sdk/index.html), `nvcc` is a CUDA compiler while `nvc` is a C11 compiler

Reply:
So `#ifndef __NVCC__` sounds like the right thing, but I'm still not clear why we can't do:
```c
#ifndef __NVCC__
#include <complex>
#else
#include <cuComplex.h>
#endif
```

Reply:
the `cuComplex` are not compatible with the C ones i think we need to do what you suggested but need to add the wrapping.
and they use their own API for the operations.
will be added in a separate issue.